server/text_generation_server/models/cache_manager.py

```diff
@@ -25,26 +25,43 @@ def __init__(
         self.repeat_slots = repeat_slots
 
         element_size = torch.tensor([], dtype=dtype).element_size()
-        if IS_XPU_SYSTEM:
-            x = 1
+        if device == torch.device("cpu"):
```

Comment:
Why ?

This breaks a lot of assumptions for other places in code.
We are experimenting with switching to flash decoding which has this layout, but it doesn't feel right to change the layout based on device.

Reply:
cpu and xpu in ipex implement difference layout for pageattention.